### DISCUSSION TITLE
Geany

### DISCUSSION THREAD

**Discussion by InfoLibre** [linuxmint/discussions#615] [Ideas]
Why not replacing xed by Geany and contributing to Geany? It would be better than continuing to waste time on xed. Geany is installed by default in Raspberry Pi OS. It’s a very more performant text editor than xed and can be used as easily as xed if you parameter it with minimalist menus.

**Comment by Jopp-gh** [comment]
Admitted, I'm not into Linux app development but use vsCode almost daily.

Geany is, as far as I can remember a very old app, so I'm wondering why people sticks to these oldies instead to switch to much better tools like the latest and much better Vscode? 
There's also a telemetry free version of vscode.
Vscode offers all kind of extensions so you can code anything with ease.
 
Anyway I agree, Geany would be a better app than xEd. Probably xed is the default, because it's so generic and ideal for simple and very quick edits or previews of text files or similar. 

I answered more for curiously than to give an answer. Still, vsCode is great.

**Reply by Secret-chest** [reply]
VSCode's UI is not native, it is web-based. The interface often behaves confusingly unlike all other apps, like with the command palette.

**Reply by Jopp-gh** [reply]
Vscode might be confusing at first, but you don't have to use all of its commands.   The more you use this IDE, the more you want to learn more, mastering Vscode, step by step. 

Also, its web-based services aren't mandatory, except for downloading plugins. So it's basically a local app and there's a telemetry free version too, otherwise I'd refrain to suggest this tool. 
  
Finally, it shouldn't matter if its framework is web-based but that's  my opinion. The interface is the same on multiple platforms, which is a bonus too.

**Reply by eht16** [reply]
> Geany is, as far as I can remember a very old app, so I'm wondering why people sticks to 

What does this say?
Geany is still developed, not as actively as we would wish but it's not dead or so. The fact that it exists since almost 20 years already, is nothing bad per se.

**Reply by waynelapierre** [reply]
VS Code is bloated as hell.

**Reply by Jopp-gh** [reply]
@waynelapierre 
You'll laugh, but once I tried to switch to Geany. 

After several days trying hard to use Geany as my default IDE I couldn't anymore, because the features it offered were limited, non existing and coding felt just more painful. 

I'm not just saying something, before I couldn't verify the opposite. For me, the tools and features in a IDE need to be comfortable, since writing, debugging etc by itself are no easy tasks. Making things more difficult as they're isn't necessary and also, not "fun". 

About the bloat, get the non telemetry version, furthermore deactivate or delete plugins you don't use. Easier than that. Just telling from my experience.

**Reply by waynelapierre** [reply]
Geany is a cross-platform alternative to Notepad++, which is highly popular among Windows users. They are something between a plain text editor and a fully featured IDE. That is a niche they try to fill.

**Reply by Secret-chest** [reply]
Geany now has LSP support.

**Comment by mtwebster** [comment]
It will be a cold day in hell before we replace any default app with something that uses Electron.

I think we looked at Geany years ago and found it difficult to turn off enough of the UI to have it default as a simple editor. I haven't tried it lately, however.

**Reply by Qronikarz** [reply]
I don't particularly care what the default text editor is gonna be in Mint because that's not a big issue, but I tried to turn off as many elements as you could in Geany since I do use it often and it is always installed on my OS.
![Zrzut ekranu z 2024-12-01 19-49-21](https://github.com/user-attachments/assets/cbfc48ea-86af-48e3-ad94-dbb547475cf8)
I just used what available in options and the view tab. The theme I use is Darcula since it's "dark" and matches the Mint look well. It's flatpak version of Geany so maybe there can be some differences if you want to build it yourself to include it in the Mint.

**Reply by Jopp-gh** [reply]
Agreed, a framework like Electron isn't ideal, also me would prefer another framework but it works, which is more important. 

Vscode might be too much for a quick editor app and for the common user too. So, I'm OK with Xed as text editor.

**Reply by Heavygrass** [reply]
The best simple editor I found is `Kate`.
No need to turn off anything in GUI and has lots of features.

**Reply by InfoLibre** [reply]
Kate and Geany are indeed the most powerful text editors. But Kate is rather nested into KDE, isn't it?

**Reply by Heavygrass** [reply]
I use it with Cinnamon all the time (did not require any special setup).
With dark theme specifically there might be things that could be improved I guess, I didn't meditate on that though.
Possibly I had to choose dark theme manually, don't remember honestly.

I tried both Kate and Geany, and Kate is much easier to use as text editor out of the box.

**Reply by Fantu** [reply]
> It will be a cold day in hell before we replace any default app with something that uses Electron.
> 
> I think we looked at Geany years ago and found it difficult to turn off enough of the UI to have it default as a simple editor. I haven't tried it lately, however.

It doesn't appear to be in electron (https://github.com/geany/geany) and it would also go against what seems to be their main goal (being lightweight)

What would you like to disable about the UI? Having a UI that's too minimal, like GNOME, seems counterproductive for desktops and workstations (it's better for small screens and touch screens) and is the reason for many forks and user migrations to other DEs.
I don't know if it's just me, but it seems to me that GNOME still retains a good percentage of users only because it's the default in the main distros. Even those who use it don't like it very much and make many additions and customizations to make it more usable.
Just as it seems to me that several core programs related to the GNOME ecosystem have deteriorated, which is why some of them have been forks, as you've done with several programs, some also seem too fragmented and have too little support.
For example, the Evince forks; I think it would have been better if there had been a single one instead of many, with the collaboration of multiple teams.

**Reply by JosephMcc** [reply]
> Having a UI that's too minimal, like GNOME, seems counterproductive for desktops and workstations

No it isn't. The idea is to have a simple text editor as part of Mint's default install. It's not about having the best tool for programming or an IDE. If a user needs that, they can install the one they prefer. Everyone likes something different.

**Reply by Fantu** [reply]
If will be disabled by default the bar on the left and below which are useful mostly in programming, debugging and some other cases of quick navigation between multiple files or structured documentation, I think it could be fine, but do not remove the menus above and make it too minimal like gnome applications, which is annoying even for novice users who make very basic use.

**Reply by Jopp-gh** [reply]
>Everyone likes something different.

Well, this sounds kind, but reality is: 
"Everyone uses software depending on skill". 

To prefer simple software as default is understandable, but simple apps shouldn't be useless,  reducing features.

**Reply by JosephMcc** [reply]
> Well, this sounds kind, but reality is:
"Everyone uses software depending on skill".

To prefer simple software as default is understandable, but simple apps shouldn't be useless, reducing features.

Your arguing for the sake of arguing. Simple apps are often designed to do one thing. They don't need a bunch of "features" that don't directly contribute to that.

**Reply by Jopp-gh** [reply]
I disagree, with the "sake of arguing" part.
Most default apps are for starters and that's nice and OK. 

Later, only some of these default apps are needed occasionally. I also call poorly designed Apps "bloat", bc they can be replaced easily, even with scripts. So for me, having less stuff on my machine is just to keep order

**Comment by eht16** [comment]
@mtwebster 
Geany has a feature to read a system-wide configuration file initally when there is no user configuration yet, by default from `/usr/local/share/geany` but for Linux Mint it is probably `/usr/share/geany/` when it uses `/usr` as prefix.

For details, see https://geany.org/manual/#global-configuration-file.

An example file to hide the sidbar, toolbar and messages window looks like:
```ini
[geany]
sidebar_visible=false
msgwindow_visible=false
pref_toolbar_show=false

color_scheme=darcula.conf
```

The Linux Mint Geany package could create this file upon installation and new users will be greeted with a very clean interface.

Maybe this is already enough?

P.S.: I personally find it a bit sad to hide so much of Geany's functionality and some users might never find the potential Geany has. But this is just my personal opinion (I'm heavily biased here :D). I do see your intention.

**Comment by InfoLibre** [comment]
Geany can open very very big files, Kate can't.
One can do a vertical text selection with Kate, if I remember.

**Reply by waynelapierre** [reply]
Exactly. Geany is basically a better alternative to Notepad++.

**Reply by spayeur** [reply]
I'm not debating very big files, but Geany has column mode. Press Shift + Alt + arrow keys to make a block selection.
